activerecord/lib/active_record/destroy_later.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  # Automatically destroy records after a specified amount of time.
+  #
+  #   # Destroy all exports 30 days after creation
+  #   class Export < ApplicationRecord
+  #     destroy_later after: 30.days
+  #   end
+  #
+  #   # Destroy exports 30 days after completing them
+  #   class Export < ApplicationRecord
+  #     destroy_later after: 30.days, if: -> { completed_previously_changed? && completed? }
```

Comment:
Added:

> `destroy_later` adds an `after_commit` callback that schedules an ActiveRecord::DestroyJob. In the example above, we use `completed_previously_changed?` to check whether the record's `completed` flag was changed in the preceding save. (`completed_changed?` wouldn't work because any prior changes to the record have been saved at the time the condition is evaluated.)

Reply:
I don’t see a great place in the API docs for `ActiveModel::Dirty` or any of the guides.

Ideally, something would explain that the `*_changed?` and `*_previously_changed?` methods are added automatically for each attribute and clearly state the difference between them. The `ActiveModel::Dirty` docs don’t do that; they’re better suited for people who are mixing it into their own (non–Active Record) classes.